### DISCUSSION TITLE
Consistent map click&drag events when clicking on pickable polygon feature

### DISCUSSION THREAD

**Discussion by thomasgwatson** [visgl/deck.gl#5387] [General]
Hey folks,

Context:
We have a bunch of polygons in a layer. They are `pickable` and have `onHover` and `onClick` callbacks. We ensure those callbacks return 'falsey' values, so that mouse events can bubble (before we did this falsey return, clicking on the features to drag the map around didn't work at all).

However, dragging the map around when clicking on those features doesn't *always* register. It's almost 60% effective to drag the map on these features (off the features it is flawless). 

Does anyone know of any gotchas that we might have missed with ensuring mouse events still bubble properly?

Sample layer here:

```
export function createThickOutlineCoverageAreaLayer(geojson, onHover, onClick, opacity = 0.2, visible = true) {
  return new GeoJsonLayer({
    id: 'thick-outline-coverage-area-layer',
    data: geojson,
    getFillColor: [82, 163, 219],
    opacity,
    getLineWidth: 4,
    pickable: true,
    visible,
    onHover: (info) => {
      onHover(info)
      return false
    },
    onClick: (event) => {
      onClick(event)
      return false
    },
    getLineColor: [82, 163, 219],
    lineWidthScale: 20,
    lineWidthMinPixels: 2,
    lineJointRounded: true
  })
}
```

**Comment by Pessimistress** [comment]
> dragging the map around when clicking on those features

By "clicking" do you mean pressing down? Does the problem go away if you remove the `onHover` and `onClick` callbacks?

**Reply by thomasgwatson** [reply]
Pressing down, sorry, yes (click and drag). The problem goes away if I remove the callbacks.

(five minutes later)

Ok, I have a suspect for what's happening. Our onHover is used to create a popup (we aren't using the Deck `getTooltip`, maybe to our peril). On some clicks and drags, it seems to be selecting the text in the popup (this was only obvious when I tried recording it in a screen cast to demo to you). To test I increased the popup's offsetY to further away from the cursor position and then it all worked.

Sorry for the run-around! I'm assuming I can either port to `getToolTip` or clean up our popup and that will resolve the issue. Thanks for your support and rely!
